p2p/net/swarm/swarm_dial.go

```diff
@@ -530,3 +530,41 @@ func isRelayAddr(addr ma.Multiaddr) bool {
 	_, err := addr.ValueForProtocol(ma.P_CIRCUIT)
 	return err == nil
 }
+
+func isWebTransport(addr ma.Multiaddr) bool {
+	_, err := addr.ValueForProtocol(ma.P_WEBTRANSPORT)
+	return err == nil
+}
+
+func isQUIC(addr ma.Multiaddr) bool {
+	_, err := addr.ValueForProtocol(ma.P_QUIC)
+	return err == nil && !isWebTransport(addr)
+}
+
+// If we have QUIC addresses, we don't want to dial WebTransport addresses.
+// It's better to have a native QUIC connection.
+// Note that this is a hack. The correct solution would be a proper
+// Happy-Eyeballs-style dialing.
+func maybeRemoveWebTransportAddrs(addrs []ma.Multiaddr) []ma.Multiaddr {
+	var hasQuic, hasWebTransport bool
+	for _, addr := range addrs {
+		if isQUIC(addr) {
+			hasQuic = true
+		}
+		if isWebTransport(addr) {
+			hasWebTransport = true
+		}
```

Comment:
```suggestion
		hasQuic = isQuic(addr)
		hasWebTransport = isWebTransport(addr)
```

Reply:
Oh wait, this is not correct. The ifs are in a loop. Will revert.